core/src/main/java/com/graphhopper/routing/subnetwork/PrepareRoutingSubnetworks.java

```diff
@@ -56,155 +68,118 @@ public PrepareRoutingSubnetworks(GraphHopperStorage ghStorage, List<PrepareJob>
-                    "The maximum number of removed edges is: " + (2 * allowedRemoved));
+        int allowedMarked = ghStorage.getEdges() / 2;
+        if (markedEdges / 2 > allowedMarked)
+            throw new IllegalStateException("Too many total (directed) edges were marked as subnetwork edges: " + markedEdges + " out of " + (2 * ghStorage.getEdges()) + "\n" +
+                    "The maximum number of subnetwork edges is: " + (2 * allowedMarked));
 
-        logger.info(jobName + " - Removed " + removedComponents + " subnetworks (biggest removed: " + biggestRemoved + " edges) -> " +
-                (ccs.getTotalComponents() - removedComponents) + " subnetwork(s) left (smallest: " + smallestRemaining + ", biggest: " + ccs.getBiggestComponent().size() + " edges)"
-                + ", total removed edges: " + removedEdgeKeys + ", took: " + sw.stop().getSeconds() + "s");
-        return removedEdgeKeys;
-    }
-
-    private int setSubnetwork(int edgeKey, EdgeFilter filter, BooleanEncodedValue inSubnetworkEnc) {
-        int edgeId = EdgeBasedTarjanSCC.getEdgeFromKey(edgeKey);
-        EdgeIteratorState edge = ghStorage.getEdgeIteratorState(edgeId, Integer.MIN_VALUE);
-        if (edgeKey % 2 == 0 && filter.accept(edge)) {
-            edge.set(inSubnetworkEnc, true);
-            return 1;
-        }
-        if (edgeKey % 2 != 0 && filter.accept(edge.detach(true))) {
-            edge.set(inSubnetworkEnc, true);
-            return 1;
-        }
-        return 0;
+        logger.info(subnetworkEnc.getName().replaceAll("_subnetwork", "") + " - Marked " + subnetworks + " subnetworks (biggest: " + biggestSubnetwork + " edges) -> " +
+                (ccs.getTotalComponents() - subnetworks) + " components(s) remain (smallest: " + smallestNonSubnetwork + ", biggest: " + ccs.getBiggestComponent().size() + " edges)"
+                + ", total marked edges: " + markedEdges + ", took: " + sw.stop().getSeconds() + "s");
+        return markedEdges;
     }
 
-    /**
-     * This method checks if the node is removed or inaccessible for ALL encoders.
-     *
-     * @return true if no edges are reachable from the specified nodeIndex for any flag encoder.
-     */
-    boolean detectNodeRemovedForAllEncoders(EdgeExplorer edgeExplorerAllEdges, int nodeIndex) {
-        // we could implement a 'fast check' for several previously marked removed nodes via GHBitSet 
-        // removedNodesPerVehicle. The problem is that we would need long-indices but BitSet only supports int (due to nodeIndex*numberOfEncoders)
+    private int setSubnetworkEdge(int edgeKey, Weighting weighting, BooleanEncodedValue subnetworkEnc) {
+        // edges that are not accessible anyway are not marked as subnetworks additionally
+        if (!Double.isFinite(weighting.calcEdgeWeightWithAccess(ghStorage.getEdgeIteratorStateForKey(edgeKey), false)))
+            return 0;
```

Comment:
I think this is useful to get a useful number of 'subnetwork edges', otherwise e.g. for bike every motorway edge would be included. But for our routing code it probably does not matter whether or not edges with access=false or infinite weight are additionally marked as subnetwork edges.